snapcraft/main.py

```diff
@@ -123,13 +130,53 @@ def main():
         common.set_target_machine(args['--target-arch'])
 
     try:
-        commands.load(cmd).main(argv=args['ARGS'])
+        run(args)
     except Exception as e:
         if args['--debug']:
             raise
 
         sys.exit(textwrap.fill(str(e)))
 
 
+def _get_lifecycle_command(args):
+    lifecycle_commands = ['pull', 'build', 'stage', 'strip']
+    lifecycle_command = [k for k in lifecycle_commands if args[k]]
+    if len(lifecycle_command) == 0:
+        return None
+    return lifecycle_command[0]
+
+
+def _get_command_from_arg(args):
+    functions = {
+        'cleanbuild': snapcraft.lifecycle.cleanbuild,
+        'init': snapcraft.lifecycle.init,
+        'login': snapcraft.login,
+        'logout': snapcraft.logout,
+        'list-plugins': _list_plugins,
+    }
+    function = [k for k in functions if args[k]]
+    if len(function) == 0:
+        return None
+    return functions[function[0]]
+
+
+def run(args):
+    lifecycle_command = _get_lifecycle_command(args)
+    argless_command = _get_command_from_arg(args)
+    if lifecycle_command:
+        snapcraft.lifecycle.execute(lifecycle_command, args['<part>'])
+    elif argless_command:
+        argless_command()
+    elif args['clean']:
+        snapcraft.lifecycle.clean(args['<part>'], args['--step'])
+    elif args['upload']:
+        snapcraft.upload(args['<snap-file>'])
+    elif args['help']:
+        snapcraft.topic_help(args['<topic>'] or args['<plugin>'],
+                             args['--devel'], args['topics'])
+    else:  # snap of default:
```

Comment:
I wanted to say "or" but "by" is better :-)

El 07/04/16 a las 12:58, Kyle Fazzari escribió:
> In snapcraft/main.py
> <https://github.com/ubuntu-core/snapcraft/pull/433#discussion_r58897251>:
> 
>> +
>> +def run(args):
>> +    lifecycle_command = _get_lifecycle_command(args)
>> +    argless_command = _get_command_from_arg(args)
>> +    if lifecycle_command:
>> +        snapcraft.lifecycle.execute(lifecycle_command, args['<part>'])
>> +    elif argless_command:
>> +        argless_command()
>> +    elif args['clean']:
>> +        snapcraft.lifecycle.clean(args['<part>'], args['--step'])
>> +    elif args['upload']:
>> +        snapcraft.upload(args['<snap-file>'])
>> +    elif args['help']:
>> +        snapcraft.topic_help(args['<topic>'] or args['<plugin>'],
>> +                             args['--devel'], args['topics'])
>> +    else:  # snap of default:
> 
> s/snap of default/snap by default/
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ubuntu-core/snapcraft/pull/433/files/71398bd40679e08f4a3233187e516894d69e7242#r58897251>
>